staging/src/k8s.io/client-go/util/jsonpath/parser.go

```diff
@@ -334,13 +334,23 @@ Loop:
 func (p *Parser) parseFilter(cur *ListNode) error {
 	p.pos += len("[?(")
 	p.consumeText()
+	quotes := 0
 Loop:
 	for {
 		switch p.next() {
 		case eof, '\n':
 			return fmt.Errorf("unterminated filter")
+		case '"':
```

Comment:
I mean maybe it could but I think this is fine, see: http://stackoverflow.com/questions/1250079/how-to-escape-single-quotes-within-single-quoted-strings